README.md
<h1 align="center">
  <a href="http://developers.mercadolibre.com/es/">
    <img src="https://user-images.githubusercontent.com/1153516/29861072-689ec57e-8d3e-11e7-8368-dd923543258f.jpg" alt="Mercado Libre Developers" width="230"></a>
  </a>
  <br>
  MercadoLibre's PHP SDK
  <br>
</h1>

<h4 align="center">This is the official PHP SDK for MercadoLibre's Platform.</h4>

<p align="center">
  <a href="https://heroku.com/deploy?template=https://github.com/Solrac5555/php-sdk">
    <img src="https://www.herokucdn.com/deploy/button.svg" alt="Deploy">
  </a>
</p>

<p align="center">
  <a href="https://heroku.com/deploy?template=https://github.com/Solrac5555/php-sdk">
    <img src="https://user-images.githubusercontent.com/1153516/29859906-9453b50c-8d3a-11e7-88b6-ab354d4a4908.png">
  </a>
</p>


## How do I install it?

       clone repository
       https://github.com/Solrac5555/php-sdk.git

## How do I use it?

The first thing to do is to instance a ```Meli``` class. You'll need to give a ```clientId``` and a ```clientSecret```. You can obtain both after creating your own application. For more information on this please read: [creating an application](http://developers.mercadolibre.com/application-manager/)

### Including the Lib
Include the lib meli in your project

```php
require '/Meli/meli.php';
```
Start the development!

### Create an instance of Meli class
Simple like this
```php
$meli = new Meli('1234', 'a secret');
```
With this instance you can start working on MercadoLibre's APIs.

There are some design considerations worth to mention.

1. This SDK is just a thin layer on top of an http client to handle all the OAuth WebServer flow for you.

2. There is JSON parsing. this SDK will include [json](http://php.net/manual/en/book.json.php) for internal usage.

3. This SDK will include [curl](http://php.net/manual/en/book.curl.php) for internal usage.

4. If you already have the access_token and the refresh_token you can pass in the constructor

```php
$meli = new Meli('1234', 'a secret', 'Access_Token', 'Refresh_Token');
```

## How do I redirect users to authorize my application?

This is a 2 step process.

First get the link to redirect the user. This is very easy! Just:

```php
$redirectUrl = $meli->getAuthUrl("http://somecallbackurl",Meli::$AUTH_URL['MLB']); //  Don't forget to change the $AUTH_URL value to match your user's Site Id.
```

This will give you the url to redirect the user. You need to specify a callback url which will be the one that the user will redirected after a successfull authrization process.

Once the user is redirected to your callback url, you'll receive in the query string, a parameter named ```code```. You'll need this for the second part of the process.

```php
$user = $meli->authorize($_GET['code'], 'http://somecallbackurl');
```

This will get an ```access_token``` and a ```refresh_token``` (is case your application has the ```offline_access```) for your application and your user.

At this stage your are ready to make call to the API on behalf of the user.

#### Making GET calls

```php
$params = array('access_token' => $access_token);
$result = $meli->get('/users/me', $params); 
 #If you wish , you can get an associative array with param $assoc = true Example:
$result = $meli->get('/users/me', $params, true); 
```

#### Making POST calls

```php
$params = array('access_token' => $access_token);

  #this body will be converted into json for you
$body = array('foo' => 'bar', 'bar' => 'foo');

$response = $meli->post('/items', $body, $params);
```

#### Making PUT calls

```php
$params = array('access_token' => $access_token);

  #this body will be converted into json for you
$body = array('foo' => 'bar', 'bar' => 'foo');

$response = $meli->put('/items', $body, $params);
```

#### Making DELETE calls
```php
$params = array('access_token' => $access_token);
$response = $meli->delete('/questions/123', $params)
```

## Examples

Don't forget to check out our examples codes in the folder [examples](https://github.com/mercadolibre/php-sdk/tree/master/examples)

## Community

You can contact us if you have questions using the standard communication channels described in the [developer's site](http://developers.mercadolibre.com/community/)

## I want to contribute!

That is great! Just fork the project in github. Create a topic branch, write some code, and add some tests for your new code.

Thanks for helping!


changelog.md
Merca4 PHP SDK (v.1.0.0)
==========================
First release


tests/meli.php
<?php
require_once '../MercadoLivre/meli.php';

class InitSDKTest extends PHPUnit_Framework_TestCase
{				

	protected static $meli;

	protected $client_id = '123';
    protected $client_secret = 'a secret';
    protected $redirect_uri = 'a redirect_uri';
    protected $access_token = 'a access_token';
    protected $refresh_token = 'a refresh_token';

    public function setUp() {

    	self::$meli = $this->getMock(
	          'Meli', array('execute'), array($this->client_id, $this->client_secret, $this->access_token, $this->refresh_token)
	        );

    }
    	#auth_url tests
		public function testGetAuthUrl() {
			$redirect_uri = self::$meli->getAuthUrl($this->redirect_uri);

			$this->assertEquals('https://auth.mercadolivre.com.mx/authorization?client_id='.$this->client_id.'&response_type=code&redirect_uri='.urlencode($this->redirect_uri), $redirect_uri);

		}

		#auth tests
		public function testAuthorize() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/oauth/token'))
             ->will($this->returnCallback('getAuthorizeMock'));

			$reponse = self::$meli->authorize('a code', $this->redirect_uri);

			$this->assertEquals(200, $reponse['httpCode']);

			$reponse = self::$meli->authorize('a code', $this->redirect_uri);

			$this->assertEquals(400, $reponse['httpCode']);

		}

		public function testRefreshAccessToken() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/oauth/token'))
             ->will($this->returnCallback('getAuthorizeMock'));

			$reponse = self::$meli->refreshAccessToken();

			$this->assertEquals(200, $reponse['httpCode']);

			$reponse = self::$meli->refreshAccessToken();

			$this->assertEquals(400, $reponse['httpCode']);

			$this->refresh_token = null;
			self::$meli = $this->getMock(
	          'Meli', array('execute'), array($this->client_id, $this->client_secret, $this->access_token, $this->refresh_token)
	        );

			$reponse = self::$meli->refreshAccessToken();

			$this->assertEquals('Offline-Access is not allowed.', $reponse['error']);
		}

		#requests tests
		public function testGet() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/sites/MLM'))
             ->will($this->returnCallback('getSimpleCurl'));
	       	
	       	$reponse = self::$meli->get('/sites/MLM');

			$this->assertEquals(200, $reponse['httpCode']);

		}

		public function testPost() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/items'))
             ->will($this->returnCallback('getSimpleCurl'));

            $body = array(
            	"condition" => "new", 
            	"warranty" => "60 dias", 
            	"currency_id" => "MEX", 
            	"accepts_mercadopago" => true, 
            	"description" => "Lindo Ray_Ban_Original_Wayfarer 555", 
            	"listing_type_id" => "bronze", 
            	"title" => "oculos Ray Ban Aviador  Que Troca As Lentes  Lancamento!", 
            	"available_quantity" => 64, 
            	"price" => 289, 
            	"subtitle" => "Acompanha 3 Pares De Lentes!! Compra 100% Segura", 
            	"buying_mode" => "buy_it_now", 
            	"category_id" => "MLM5125", 
            	"pictures" => array(
            		array(
            			"source" => "http://upload.wikimedia.org/wikipedia/commons/f/fd/Ray_Ban_Original_Wayfarer.jpg"
            		), 
            		array(
            			"source" => "http://en.wikipedia.org/wiki/File:Teashades.gif"
            		)
            	)
            );

            $params = array('access_token' => $this->access_token);
	       	
	       	$reponse = self::$meli->post('/items', $body, $params);

			$this->assertEquals(200, $reponse['httpCode']);

		}

		public function testPut() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/items/MLM123'))
             ->will($this->returnCallback('getSimpleCurl'));

            $body = array(
            	"available_quantity" => 10, 
            	"price" => 280, 
            );

            $params = array('access_token' => $this->access_token);
	       	
	       	$reponse = self::$meli->put('/items/MLM123', $body, $params);

			$this->assertEquals(200, $reponse['httpCode']);

		}

		public function testDelete() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/questions/123'))
             ->will($this->returnCallback('getSimpleCurl'));

            $params = array('access_token' => $this->access_token);
	       	
	       	$reponse = self::$meli->delete('/questions/123', $params);

			$this->assertEquals(200, $reponse['httpCode']);

		}

		public function testOptions() {
			self::$meli->expects($this->any())
             ->method('execute')
             ->with($this->equalTo('/sites/MLM'))
             ->will($this->returnCallback('getSimpleCurl'));
	       	
	       	$reponse = self::$meli->options('/sites/MLM');

			$this->assertEquals(200, $reponse['httpCode']);

		}

		#makePath tests
		public function testMakePath() {
			$params = array(
				'access_token' => 'a access_token',
				'ids' => 'MLM123,MLM321'
			);

	       	$reponse = self::$meli->make_path('/items', $params);
			
			$this->assertEquals('https://api.mercadolibre.com/items?access_token=a access_token&ids=MLM123,MLM321', $reponse);

			$reponse = self::$meli->make_path('items', $params);
			
			$this->assertEquals('https://api.mercadolibre.com/items?access_token=a access_token&ids=MLM123,MLM321', $reponse);

			// $reponse = self::$meli->make_path('https://api.mercadolibre.com/items', $params);
			
			// $this->assertEquals('https://api.mercadolibre.com/items?access_token=a access_token&ids=MLB123,MLB321', $reponse);
			
		}

		
    public function tearDown() {
		parent::tearDown();
    }
}


#Mock requests
$code = 0;
function getAuthorizeMock() {
	global $code;
	$code++;

	if($code == 1) {
		$body = array(
			'access_token' => 'a access_token', 
			'token_type' => 'bearer',
			'expires_in' => '10800',
			'scope' => 'offline_access read write',
			'refresh_token' => 'a refresh_token' 
		);
		$return['body'] = (object) $body;
		$return['httpCode'] = 200;
	} else {
		$body = array(
			'message' => 'Error validating grant. Your authorization code or refresh token may be expired or it was already used.',
			'error' => 'invalid_grant',
			'status' => 400,
			'cause' => array()
		);
		$return['body'] = (object) $body;
		$return['httpCode'] = 400;
	}
	
	return $return;
}


$refresh = 0;
function getRefreshTokenMock() {
	global $refresh;
	$refresh++;

	if($refresh == 1) {
		$body = array(
			'access_token' => 'a access_token', 
			'token_type' => 'bearer',
			'expires_in' => '10800',
			'scope' => 'offline_access read write',
			'refresh_token' => 'a refresh_token' 
		);
		$return['body'] = (object) $body;
		$return['httpCode'] = 200;
	} else if($refresh == 2) {
		$body = array(
			'message' => 'Error validating grant. Your authorization code or refresh token may be expired or it was already used.',
			'error' => 'invalid_grant',
			'status' => 400,
			'cause' => array()
		);
		$return['body'] = (object) $body;
		$return['httpCode'] = 400;
	} else {
		$body = array(
			'message' => 'Error validating grant. Your authorization code or refresh token may be expired or it was already used.',
			'error' => 'invalid_grant',
			'status' => 400,
			'cause' => array()
		);
		$return['body'] = (object) $body;
		$return['httpCode'] = 400;
	}
	
	return $return;
}

function getSimpleCurl() {
	$return['body'] = 'null';
	$return['httpCode'] = 200;

	return $return;
}



examples/example_list_item.php
<?php
session_start();
require '../Meli/meli.php';
require '../configApp.php';
$meli = new Meli($appId, $secretKey);
if($_GET['code']) {
	// If the code was in get parameter we authorize
	$user = $meli->authorize($_GET['code'], $redirectURI);
	// Now we create the sessions with the authenticated user
	$_SESSION['access_token'] = $user['body']->access_token;
	$_SESSION['expires_in'] = $user['body']->expires_in;
	$_SESSION['refresh_token'] = $user['body']->refresh_token;
	// We can check if the access token in invalid checking the time
	if($_SESSION['expires_in'] + time() + 1 < time()) {
		try {
			print_r($meli->refreshAccessToken());
		} catch (Exception $e) {
			echo "Exception: ",  $e->getMessage(), "\n Line 18";
		}
	}
	// We construct the item to POST
	$item = array(
		"title" => "Item de prueva favor no ofertar! --kc:off",
        "category_id" => "MLA1051",
        "price" => 10,
        "currency_id" => "MXN",
        "available_quantity" => 1,
        "buying_mode" => "buy_it_now",
        "listing_type_id" => "bronze",
        "condition" => "new",
        "description" => array ("plain_text" => "Item de Teste. Mercado Livre's PHP SDK."),
        "video_id" => "RXWn6kftTHY",
        "warranty" => "12 month",
        "pictures" => array(
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/6/64/IPhone_7_Plus_Jet_Black.svg/440px-IPhone_7_Plus_Jet_Black.svg.png"
            ),
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/IPhone7.jpg/440px-IPhone7.jpg"
            )
        ),
        "attributes" => array(
            array(
                "id" => "EAN",
                "value_name" => "190198043566"
            ),
            array(
                "id" => "COLOR",
                "value_id" => "52049"
            ),
            array(
                "id" => "WEIGHT",
                "value_name" => "188g"
            ),
            array(
                "id" => "SCREEN_SIZE",
                "value_name" => "4.7 polegadas"
            ),
            array(
                "id" => "TOUCH_SCREEN",
                "value_id" => "242085"
            ),
            array(
                "id" => "DIGITAL_CAMERA",
                "value_id" => "242085"
            ),
            array(
                "id" => "GPS",
                "value_id" => "242085"
            ),
            array(
                "id" => "MP3",
                "value_id" => "242085"
            ),
            array(
                "id" => "OPERATING_SYSTEM",
                "value_id" => "296859"
            ),
            array(
                "id" => "OPERATING_SYSTEM_VERSION",
                "value_id" => "iOS 10"
            ),
            array(
                "id" => "DISPLAY_RESOLUTION",
                "value_id" => "1920 x 1080"
            ),
            array(
                "id" => "BATTERY_CAPACITY",
                "value_name" => "3980 mAh"
            ),
            array(
                "id" => "FRONT_CAMERA_RESOLUTION",
                "value_name" => "7 mpx"
            )
        )
    );
	
	// We call the post request to list a item
	echo '<pre>';
	print_r($meli->post('/items', $item, array('access_token' => $_SESSION['access_token'])));
	echo '</pre>';
} else {
	echo '<a href="' . $meli->getAuthUrl($redirectURI, Meli::$AUTH_URL['MLM']) . '">Login using MercadoLibre oAuth XXXX 2.0</a>';
}


index.php
<?php
session_start();
require 'Meli/meli.php';
require 'configApp.php';
$domain = $_SERVER['HTTP_HOST'];
$appName = explode('.', $domain)[0];
?>

    <!DOCTYPE html>
    <html lang="en" class="no-js">

    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width,initial-scale=1">
        <meta name="description" content="My PHP SDK for Mercado Libre's API.">
        <meta name="keywords" content="API, PHP, Mercado Libre, SDK, meli, integration, e-commerce">
        <title>Mercado Libre PHP SDK</title>
        <link rel="stylesheet" href="/getting-started/style.css" />
        <script src="script.js"></script>
    </head>

    <body>
        <header class="navbar">
            <a class="logo" href="#"><img src="https://www.ingeniatv.com/wp-content/uploads/2018/07/IngTv9.png" alt="" width="324" height="90"></a>
            <nav>
                <ul class="nav navbar-nav navbar-right">
                    <li><a target="_blank" href="http://developers.mercadolibre.com/getting-started/">Getting Started</a></li>
                    <li><a target="_blank" href="http://developers.mercadolibre.com/api-docs/">API Docs</a></li>
                    <li><a target="_blank" href="http://developers.mercadolibre.com/community/">Community</a></li>
                </ul>
            </nav>
        </header>
        <div class="header">
            <div>
                <h1>API ML PHP SDK</h1>
                <h2>My PHP SDK for Mercado Libre's API.</h2>
            </div>
        </div>
        <main class="container">
            <h3>Hola,</h3>
            <p>This is a sample app, deployed to Heroku with Mercado Libre's PHP SDK. Feel free to use it as a base, to start building your awesome app!</p>
            <div class="row">
                <div class="col-md-6">
                    <h3>How it works?</h3>
                    <ul>
                        <li>This app was deployed to Heroku, either using Git or by using <a href="https://github.com/heroku/go-getting-started">Heroku Button</a> on the repository.</li>
                        <li>When Heroku received the source code it used the go tool chain to compile the application along with any vendored dependencies and created a deployable slug.</li>
                        <li>The platform then spins up a dyno, a lightweight container that provides an isolated environment in which the slug can be mounted and executed.</li>
                        <li>You can scale your app, manage it, and deploy over <a href="https://addons.heroku.com/">150 add-on services</a>, from the Dashboard or CLI.</li>
                    </ul>
                </div>
                <div class="col-md-6">
                    <h3>Next steps</h3>
                    <p>To start, <a href="https://developers.mercadolibre.com.ar/apps/home">go to your My Apps dashboard</a> and update your application's <b>redirect URI</b> to match the one Heroku is running: <br />
                        <code><?php echo 'https://'.$domain; ?></code>.
                        <br />
                        <br /> If you deployed this app by the Heroku Button, you need to clone this aplication to your computer by running the following on a command line shell:
                        <br />
                        <code>heroku git:clone -a <?php echo $appName; ?></code>
                        <br /> This will create a local copy of the source code, and associate the Heroku app with your local repository.</p>
                    <p>Follow the offical Heroku's guide <a target="_blank" href="https://devcenter.heroku.com/articles/git">https://devcenter.heroku.com/articles/git</a> to deploy using the Heroku cli.</p>
                    <p>You'll now be set up to run the app locally, or deploy changes to Heroku.</p>
                </div>
            </div>
            <div class="row">
                <h3>Examples</h3>
                <p>
                    Check the following examples, they will help you to start working with our API!
                </p>
                <p>
                    Note that these examples work for MLM(México) by default. If you'd like to try them in your own country, please, <a href="https://github.com/Solrac5555/php-sdk/blob/master/configApp.php">update this line</a> in your project, with
                    your own <b>$site_id</b> before executing them.
                </p>
            </div>
            <hr>
            <div class="row">
                <div class="col-sm-6 col-md-6">
                    <h3>oAuth</h3>
                    <p>First authenticate yourself. Authentication is the key to get the most ouf Mercado Libre's API.</p>
                    <?php
                    $meli = new Meli($appId, $secretKey);
                    if($_GET['code'] || $_SESSION['access_token']) {
                        // If code exist and session is empty
                        if($_GET['code'] && !($_SESSION['access_token'])) {
                            // If the code was in get parameter we authorize
                            $user = $meli->authorize($_GET['code'], $redirectURI);
                            // Now we create the sessions with the authenticated user
                            $_SESSION['access_token'] = $user['body']->access_token;
                            $_SESSION['expires_in'] = time() + $user['body']->expires_in;
                            $_SESSION['refresh_token'] = $user['body']->refresh_token;
                        } else {
                            // We can check if the access token in invalid checking the time
                            if($_SESSION['expires_in'] < time()) {
                                try {
                                    // Make the refresh proccess
                                    $refresh = $meli->refreshAccessToken();
                                    // Now we create the sessions with the new parameters
                                    $_SESSION['access_token'] = $refresh['body']->access_token;
                                    $_SESSION['expires_in'] = time() + $refresh['body']->expires_in;
                                    $_SESSION['refresh_token'] = $refresh['body']->refresh_token;
                                } catch (Exception $e) {
                                    echo "Exception: ",  $e->getMessage(), "\n";
                                }
                            }
                        }
                        echo '<pre>';
                            print_r($_SESSION);
                        echo '</pre>';
                    } else {
                        echo '<p><a alt="Login using MercadoLibre oAuth 2.0" class="btn" href="' . $meli->getAuthUrl($redirectURI, Meli::$AUTH_URL[$siteId]) . '">Authenticate</a></p>';
                    }
                    ?>
                </div>
                <div class="col-sm-6 col-md-6">
                    <h3>Get site</h3>
                    <p>Make a simple GET to <a href="https://api.mercadolibre.com/sites">sites resource</a> with your <b>$site_id</b> to obtain information about a a site. Like id, name, currencies, categories, and other settings.</p>
                    <p><a class="btn" href="../examples/example_get.php">GET</a></p>
                </div>
            </div>
            <hr>
            <div class="row">
                <div class="col-md-6">
                    <h3>Publish an Item</h3>
                    <p>
                        This is a example of how to list an item in <b>MLM</b> (México).
                       <br /> <b>You need to be authenticated to make it work.</b>
                       <br /> To be able to list an item in another country, <a href="https://github.com/Solrac5555/php-sdk/blob/master/examples/example_list_item.php">please update this file</a>, with values according to the site Id where your app works, like <b>category_id</b> and <b>currency</b>.
                     <br />
                    </p>
                    <pre class="pre-item">
				"title" => "Item de prueba favor no ofertar! --kc:off",
        "category_id" => "MLA1051",
        "price" => 10,
        "currency_id" => "MEX",
        "available_quantity" => 1,
        "buying_mode" => "buy_it_now",
        "listing_type_id" => "bronze",
        "condition" => "new",
        "description" => array ("plain_text" => "Item de Teste. Mercado Livre's PHP SDK."),
        "video_id" => "RXWn6kftTHY",
        "warranty" => "12 month",
        "pictures" => array(
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/6/64/IPhone_7_Plus_Jet_Black.svg/440px-IPhone_7_Plus_Jet_Black.svg.png"
            ),
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/IPhone7.jpg/440px-IPhone7.jpg"
            )
        )
    )
                    </pre>
                    <?php
                    $meli = new Meli($appId, $secretKey);
                    if($_GET['code'] && $_GET['publish_item']) {
                        // If the code was in get parameter we authorize
                        $user = $meli->authorize($_GET['code'], $redirectURI);
                        // Now we create the sessions with the authenticated user
                        $_SESSION['access_token'] = $user['body']->access_token;
                        $_SESSION['expires_in'] = $user['body']->expires_in;
                        $_SESSION['refresh_token'] = $user['body']->refresh_token;
                        // We can check if the access token in invalid checking the time
                        if($_SESSION['expires_in'] + time() + 1 < time()) {
                            try {
                                print_r($meli->refreshAccessToken());
                            } catch (Exception $e) {
                                echo "Exception: ",  $e->getMessage(), "Salto de line \n";
                            }
                        }
                        // We construct the item to POST
						
						$item = array(
		"title" => "Item de prueba favor no ofertar! --kc:off",
        "category_id" => "MLA1051",
        "price" => 10,
        "currency_id" => "MEX",
        "available_quantity" => 1,
        "buying_mode" => "buy_it_now",
        "listing_type_id" => "bronze",
        "condition" => "new",
        "description" => array ("plain_text" => "Item de Teste. Mercado Livre's PHP SDK."),
        "video_id" => "RXWn6kftTHY",
        "warranty" => "12 month",
        "pictures" => array(
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/6/64/IPhone_7_Plus_Jet_Black.svg/440px-IPhone_7_Plus_Jet_Black.svg.png"
            ),
            array(
                "source" => "https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/IPhone7.jpg/440px-IPhone7.jpg"
            )
        ),
        "attributes" => array(
            array(
                "id" => "EAN",
                "value_name" => "190198043566"
            ),
            array(
                "id" => "COLOR",
                "value_id" => "52049"
            ),
            array(
                "id" => "WEIGHT",
                "value_name" => "188g"
            ),
            array(
                "id" => "SCREEN_SIZE",
                "value_name" => "4.7 polegadas"
            ),
            array(
                "id" => "TOUCH_SCREEN",
                "value_id" => "242085"
            ),
            array(
                "id" => "DIGITAL_CAMERA",
                "value_id" => "242085"
            ),
            array(
                "id" => "GPS",
                "value_id" => "242085"
            ),
            array(
                "id" => "MP3",
                "value_id" => "242085"
            ),
            array(
                "id" => "OPERATING_SYSTEM",
                "value_id" => "296859"
            ),
            array(
                "id" => "OPERATING_SYSTEM_VERSION",
                "value_id" => "iOS 10"
            ),
            array(
                "id" => "DISPLAY_RESOLUTION",
                "value_id" => "1920 x 1080"
            ),
            array(
                "id" => "BATTERY_CAPACITY",
                "value_name" => "3980 mAh"
            ),
            array(
                "id" => "FRONT_CAMERA_RESOLUTION",
                "value_name" => "7 mpx"
            )
        )
    );                    
                        $response = $meli->post('/items/', $item, array('access_token' => $_SESSION['access_token']));
                        // We call the post request to list a item
                        echo "<h4>Response</h4>";
                        echo '<pre class="pre-item">';
                        print_r ($response);
                        echo '</pre>';
                        echo "<h4>Success! Your test item was listed!</h4>";
                        echo "<p>Go to the permalink to see how it's looking in our site.</p>";
                        echo '<a target="_blank" href="'.$response["body"]->permalink.'">'.$response["body"]->permalink.'</a><br />';
                    } else if($_GET['code']) {
                        echo '<p><a alt="Publish Item" class="btn" href="/?code='.$_GET['code'].'&publish_item=ok">Publish Item</a></p>';
                    } else {
                        echo '<p><a alt="Publish Item" class="btn disable" href="#">Publish Item</a> </p>';
                    }
                    ?>
                </div>
                <div class="col-md-6">
                    <h3>Get started!</h3>
                    <p>Now you know how easy it is to get information from our API. Check the rest of the examples on the SDK, and modify them as you like in order to List an item, update it, and other actions.</p>
                    <p><a class="btn" href="https://github.com/Solrac5555/php-sdk/tree/master/examples">More examples</a></p>
                </div>
            </div>
            <hr>
            <div class="row">
                <h3>Tus Datos</h3>
                <div class="row-info col-sm-3 col-md-3">
                    <b>App_Id: </b>
                    <?php echo $appId; ?>
                </div>
                <div class="row-info col-sm-3 col-md-3">
                    <b>Secret_Key: </b>
                    <?php echo $secretKey; ?>
                </div>
                <div class="row-info col-sm-3 col-md-3">
                    <b>Redirect_URI: </b>
                    <?php echo $redirectURI; ?>
                </div>
                <div class="row-info col-sm-3 col-md-3">
                    <b>Site_Id: </b>
                    <?php echo $siteId; ?>
                </div>
            </div>
            <hr>
        </main>
    </body>
    </html>
